From ec372b738b022607e02a1fb1032a9d8be796a9b7 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Fri, 12 Aug 2022 17:03:54 +0800
Subject: [PATCH 15/46] Revert "drm: msm: fix possible memory leak in
 mdp5_crtc_cursor_set()"

This reverts commit 449374565f349d4233beec811d4286fdfe5de44b.
---
 drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c b/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c
index 03d60eb09257..9afbce3cb87b 100644
--- a/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c
+++ b/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c
@@ -909,10 +909,8 @@ static int mdp5_crtc_cursor_set(struct drm_crtc *crtc,
 
 	ret = msm_gem_get_and_pin_iova(cursor_bo, kms->aspace,
 			&mdp5_crtc->cursor.iova);
-	if (ret) {
-		drm_gem_object_put(cursor_bo);
+	if (ret)
 		return -EINVAL;
-	}
 
 	pm_runtime_get_sync(&pdev->dev);
 
-- 
2.27.0

